fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -91,6 +91,15 @@ ThreadFuture<Void> ThreadSafeDatabase::createSnapshot(const StringRef& uid, cons
 	return onMainThread([db, snapUID, cmd]() -> Future<Void> { return db->createSnapshot(snapUID, cmd); });
 }
 
+double ThreadSafeDatabase::getMainThreadBusyness() {
+	// Return the main network thread busyness
+	if (!g_network) {
+		// TODO: Is this the right thing to do in this case?
+		return 0.0;
```

Comment:
Another option here would be to have the network busyness API return an `fdb_error_t` and return the actual busyness through an output parameter. Then the client could check the error return to see if the value is valid, much like many of the other C functions. On the other hand, this is a pretty simple API call that should pretty much always be valid unless the network thread has not been started. In that case, maybe it would even be ok to throw an error here.

Reply:
Taking another look at the code it seems that in NativeAPI we will create [g_network](https://github.com/apple/foundationdb/blob/5f079904d40543f642c0b85df4cd7025d7bcd420/fdbclient/NativeAPI.actor.cpp#L1745) before starting the monitor, so this check might not even be necessary. @sfc-gh-ajbeamon what do you think?

Reply:
The database can't be created until after the network is setup, so I don't think you can call this function with no `g_network` and this block wouldn't be needed.

Reply:
My reading of https://apple.github.io/foundationdb/api-c.html#c.fdb_setup_network is that there aren't thread safety issues accessing g_network here (as long as the client obeys the contract /shrug).

I would consider an `ASSERT(g_network)` instead

Reply:
sounds good i'll add the assert thanks

Reply:
It looks like creating a database on a network that isn't setup also throws an error, so that even protects you if you don't follow the contract.

https://github.com/apple/foundationdb/blob/e7abffbe71c88a9b116392f179e733629411f217/fdbclient/MultiVersionTransaction.actor.cpp#L1564